src-v2/Judge/NTS.Judge.Blazor/Pages/Dashboard/Components/Actions/EliminationForms/FailedToQualifyForm.razor

```diff
@@ -4,18 +4,19 @@
 @using NTS.Judge.Blazor.Ports
 @using Not.Structures
 
-@inherits NotComponent
+@inherits EliminationForm
 
-<MudSelect T="string" Label="Failed to Qualify Reason" MultiSelection="true" Delimiter="+" SelectedValues="@FTQReason" SelectedValuesChanged="SetInputState" Placeholder="Select a reason">
-    @foreach (var failToQualifyCode in SelectListModel.SelectListFromEnum<FTQCodes>())
+<MudSelect T="FTQCodes" Label="Failed to Qualify Reason" MultiSelection="true" Delimiter="+"
+    SelectedValues="@Codes" SelectedValuesChanged="SetInputState" Placeholder="Select a reason">
+    @foreach (var failToQualifyCode in SelectListModel.FromEnum<FTQCodes>())
     {
         <MudSelectItem T="string" Value="@failToQualifyCode.Value">
```

Comment:
Honestly I'm surprised this is working now, because I would expect that MudSelect and MudSelectItem should have the same type of T. Anyhow, doesn't it make more sense to have the value of SelectListModel to be int. Enum types are int by default, so they would be interchangeable.

Furthermore you should be able to remove T property here as well

Reply:
"Honestly I'm surprised this is working now" - it's no longer working when I update to dotnet 8. It's causing the application to hang on a white screen with no errors whatsoever. You can see how I've fixed it [here](https://github.com/Not-Endurance/not-timing-system/commit/c9c52e5a5cd74b2f7924ef34927ccf313d2bee62)

Reply:
Don't fix it in this PR.